crm-api/src/main/java/com/agilemonkeys/test/crm/server/resource/service/DataBaseCustomerService.java

```diff
@@ -0,0 +1,125 @@
+package com.agilemonkeys.test.crm.server.resource.service;
+
+import com.agilemonkeys.test.crm.commons.exception.EntityNotFoundCRMException;
+import com.agilemonkeys.test.crm.server.resource.model.dto.CustomerDto;
+import com.agilemonkeys.test.crm.server.resource.model.dto.PhotoDto;
+import com.agilemonkeys.test.crm.server.resource.model.dto.VersionDto;
+import com.agilemonkeys.test.crm.server.resource.model.entity.Customer;
+import com.agilemonkeys.test.crm.server.resource.repository.CustomerRepository;
+import com.agilemonkeys.test.crm.server.resource.util.ModelMapperUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.Assert;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.ResourceBundle;
+import java.util.UUID;
+
+@Service
+public class DataBaseCustomerService  implements CustomerService  {
+
+    @Autowired
+    CustomerRepository customerRepository;
+
+    @Autowired
+    ModelMapperUtil modelMapper;
+
+    @Override
+    public CustomerDto createOrUpdateCustomer(CustomerDto customerDto) {
+        Customer customerSaved;
+        Customer customer = customerRepository.findOne(customerDto.getId());
+        if (customer != null && customer.getId() != null) {
+            customer.setName(customerDto.getName());
+            customer.setSurname(customerDto.getSurname());
+            customerSaved =customerRepository.save(customer);
+        } else {
+            customer = modelMapper.map(customerDto,Customer.class);
+            customerSaved =customerRepository.save(customer);
+
+        }
+        return modelMapper.map(customerSaved, CustomerDto.class);
+    }
+
+    @Override
+    public void deleteCustomer (String idCustomer) {
+        customerRepository.delete(idCustomer);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
```

Comment:
Yes, it´s only query, but with the annotation "readOnly" the performance it´s better because in the method FLUSH_METHOD is never and depend of the implementations of JPA has mor performance(for example in Oracle)